components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/CacheBackedAuthenticatorMgtDAO.java

```diff
@@ -53,9 +58,12 @@ public UserDefinedLocalAuthenticatorConfig addUserDefinedLocalAuthenticator(
 
         AuthenticatorCacheKey cacheKey = new AuthenticatorCacheKey(authenticatorConfig.getName());
         authenticatorCache.addToCache(cacheKey, new AuthenticatorCacheEntry(createdConfig), tenantId);
+        userDefinedLocalAuthenticatorsCache.clearCacheEntry(new UserDefinedLocalAuthenticatorsCacheKey(tenantId),
+                tenantId);
         if (LOG.isDebugEnabled()) {
             LOG.debug(String.format(
                     "Adding cache entry for newly created authenticator %s.", authenticatorConfig.getName()));
+            LOG.debug("Delete Cache entry for all user defined authenticators of tenant id: " + tenantId);
```

Comment:
user defined local authenticators. Fix the latter similarly

Reply:
Fixed